core/types/transaction_marshalling.go

```diff
@@ -324,6 +326,12 @@ func (tx *Transaction) UnmarshalJSON(input []byte) error {
 		if dec.IsSystemTx != nil {
 			itx.IsSystemTransaction = *dec.IsSystemTx
 		}
+
+		if dec.Nonce != nil {
+			// TODO: this works but feels pretty ick...
+			itxNonce := depositTxWithNonce{DepositTx: itx, Nonce: uint64(*dec.Nonce)}
+			inner = &itxNonce
+		}
```

Comment:
See other comment, I think we can just drop the `dec.Nonce` here and not introduce the inner-type wrapper, and not put it anywhere, as it'll already be available via the receipt (and trusted that way)